presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DesugaringOptimizer.java

```diff
@@ -194,15 +189,8 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<Void> context)
 
         private Expression desugar(Expression expression)
         {
-            checkState(!(expression instanceof GroupingOperation), "GroupingOperation should have been re-written to a FunctionCall before execution");
-
-            if (expression instanceof SymbolReference) {
-                return expression;
-            }
```

Comment:
I think that above `if` and `checkState` should remain here. `if` and `checkState` apply to both rewrites,.